prolog/comment.pl

```diff
@@ -134,23 +140,79 @@
     test(comment_token_rec('<','>',TT),S,[]),
     atom_codes(T,TT).
 
-tester([]).
-tester([X|L]) :-
-    write_term(test(X),[]),
+start(AA) :-
+    (
+        catch(b_getval(a,[N,A]),_,N=0) ->
+          true;
+        N=0
+    ),
+    NN is N + 1,
     (
-        call(X) -> write(' ... OK') ; write(' ... FAIL')
+        N == 0 ->
+          AA = _;
+        AA = A
     ),
-    nl,
-    tester(L).
+    b_setval(a,[NN,AA]).
 
+end(A) :-
+    b_getval(a,[N,A]),
+    NN is N - 1,
+    b_setval(a,[NN,A]).
+    
+a(A) --> 
+  {atom_codes(A,AA)},
+   AA,
+   {start(B)},
+   [B].
+
+a(A,C) --> 
+  {atom_codes(A,AA)},
+   AA,
+   {start(B)},
+   [B],
+   {append(AA,[B],C)}.
+
+b(A) -->
+ {end(B)},
+ [B],
+ {atom_codes(A,AA)},
+  AA.
+
+b(A,C) -->
+ {end(B)},
+ [B],
+ {atom_codes(A,AA)},
+  AA,
+  {append([B],AA,C)}.
+
+test_adapt(S,T) :-
+    test(comment_token_rec(a('<'),b('>'),TT),S,[]),
+    atom_codes(T,TT).
+    
 
-/*
-tester(
-    [test_comment('<alla>'),
-     test_comment_rec('<alla<balla>>'),
-     test_comment_token('<alla>','<alla>'),
-     test_comment_token_rec('<alla<balla>>','<alla<balla>>')]).
-*/
+:- multifile test/2.
+             
+test('Test comment',[true(test_comment('<alla>'))]) :- true.
+test('Test comment_rec',[true(test_comment_rec('<alla<balla>>'))]) :- true.
+test('Test comment_token',
+     [true(A==B)]) :- 
```

Comment:
Spaces around operators, please.

```suggestion
     [true(A == B)]) :- 
```